kratos/processes/simple_mortar_mapper_process.h

```diff
@@ -0,0 +1,474 @@
+    
+        // Create the mortar operators
+        MortarOperator<TNumNodes> rThisMortarConditionMatrices;
+        
+        // We call the exact integration utility
+        ExactMortarIntegrationUtility<TDim, TNumNodes> integration_utility = ExactMortarIntegrationUtility<TDim, TNumNodes>(TDim);
+        
+        // Initialize to zero // TODO: Add OMP
+        for(int i=0; i<static_cast<int>(mrThisModelPart.Conditions().size()); i++)
+        {
+            auto it_cond = mrThisModelPart.ConditionsBegin() + i;
+            
+            if (it_cond->Is(SLAVE) == true)
+            {
+                boost::shared_ptr<ConditionMap>& all_conditions_maps = it_cond->GetValue( MAPPING_PAIRS );
+                
+                for (auto it_pair = all_conditions_maps->begin(); it_pair != all_conditions_maps->end(); ++it_pair )
+                {
+                    Condition::Pointer p_cond_master = (it_pair->first); // MASTER
+                    GeometryType& master_geometry = p_cond_master->GetGeometry();
+                    
+                    for (unsigned int i_node = 0; i_node < TNumNodes; i_node++)
+                    {
+                        MortarUtilities::ResetValue<TVarType, THist>(master_geometry[i_node], mDestinyVariable);
+                    }
+                }
+            }
+        }
+        
+        // We map the values from one side to the other // TODO: Add OMP
+        for(int i=0; i<static_cast<int>(mrThisModelPart.Conditions().size()); i++)
+        {
+            auto it_cond = mrThisModelPart.ConditionsBegin() + i;
+            
+            if (it_cond->Is(SLAVE) == true)
+            {
+                const array_1d<double, 3>& slave_normal = it_cond->GetValue(NORMAL);
+                GeometryType& slave_geometry = it_cond->GetGeometry();
+                
+                boost::shared_ptr<ConditionMap>& all_conditions_maps = it_cond->GetValue( MAPPING_PAIRS );
+                
+                for (auto it_pair = all_conditions_maps->begin(); it_pair != all_conditions_maps->end(); ++it_pair )
+                {
+                    Condition::Pointer p_cond_master = (it_pair->first); // MASTER
+                    const array_1d<double, 3>& master_normal = p_cond_master->GetValue(NORMAL); 
+                    GeometryType& master_geometry = p_cond_master->GetGeometry();
+                        
+                    const double total_master_area = master_geometry.Area();
+                    double master_area = 0.0;
+                    
+                    IntegrationMethod this_integration_method = GeometryData::GI_GAUSS_2;
+                    
+                    // Reading integration points
+                    std::vector<array_1d<PointType,TDim>> conditions_points_slave;
+                    const bool is_inside = integration_utility.GetExactIntegration(slave_geometry, slave_normal, master_geometry, master_normal, conditions_points_slave);
```

Comment:
how do you internally select whether they are overlapping or not?
Are you doing some projection operations?

Reply:
For this probably the documentation of the IntegrationUtility would help :D

Reply:
Yes, there are projection operations. What do you mean with overlapping?

Reply:
Overlapping in the sense that the master and the slave conditions are in the same location geometrically. You can only integrate if the conditions are overlapping

I was asking abt the projection bcs I saw that you do it later again. That one however is then to get the SFvalues of the master or slave from what I could see